jsapp/js/components/processing/singleProcessingStore.ts

```diff
@@ -936,6 +983,63 @@ class SingleProcessingStore extends Reflux.Store {
       this.isProcessingDataLoaded
     );
   }
+
+  getDisplays() {
+    return this.displays;
+  }
+
+  refreshDisplays() {
+    const tab = this.getActiveTab();
+
+    // Tab specific displays
+    if (tab === SingleProcessingTabs.Translations) {
+      this.displays.set(StaticDisplays.Transcript, true);
+      this.clearTranslationDisplays();
+    } else if (tab === SingleProcessingTabs.Transcript) {
+      this.displays.delete(StaticDisplays.Transcript);
+      this.populuateTranslationDisplays();
+    } else {
+      this.populuateTranslationDisplays();
+      // If transcript display is on (coming from translations) then leave it on.
+      this.displays.set(
+        StaticDisplays.Transcript,
+        this.displays.get(StaticDisplays.Transcript) || false
+      );
+    }
+  }
+
+  setStaticDisplay(display: StaticDisplays) {
```

Comment:
I think it would be cleaner if instead of display name, we would also give a boolean `isEnabled` here, so `setStaticDisplay(display: StaticDisplays, isEnabled: boolean) {`. The `<ToggleSwitch>`es that uses this function would use their boolean callback value :)